### ISSUE TITLE
Compile-AppInBcContainer does not throw error on missing RDLC

### ISSUE THREAD

**Issue by michvllni** [microsoft/navcontainerhelper#1696]
**Describe the issue**
`Compile-AppInBcContainer` usually throws the error `app generation failed` when there are errors while compiling the app.
However, this does not happen when I'm trying to compile an app where an RDLC file is referenced but not present while GenerateRDLCLayout is set to "No". 

alc.exe throws the error, but it gets printed AFTER the compiler writes "compilation ended at XX:XX:XX".

I think the issues appears because alc.exe creates an app file even though the error occurs.

This is an issue for me because we check if there were errors in the compile to validate the build.

As a workaround i think i could check the output of `Compile-AppinBcContainer` for `System\.IO\.FileNotFoundException: Could not find file '.*\.rdlc'` and throw an error if it is found

**Scripts used**
```
Compile-AppInBcContainer -containerName DEV `
-appProjectFolder "C:\sandbox" `
-appOutputFolder "C:\sandbox\output" `
-GenerateReportLayout "No" `
-credential $cred
```
**Full output of scripts**
```
Using Symbols Folder: E:\Repositories\developer-sandbox\.alpackages
Compiling...
.\alc.exe /project:"C:\sandbox" /packagecachepath:"C:\sandbox\.alpackages" /out:"C:\sandbox\output\appfile.app" /GenerateReportLayout- /assemblyprobingpaths:"C:\Program Files (x86)\Microsoft Dynamics NAV\160\RoleTailored Client","C:\Program Files\Microsoft Dynamics NAV\160\Service","C:\Program Files (x86)\Open XML SDK\V2.5\lib","c:\Windows\Microsoft.NET\Assembly","C:\Test Assemblies\Mock Assemblies"
Microsoft (R) AL Compiler version 5.0.4.26576
Copyright (C) Microsoft Corporation. All rights reserved

Compilation started for project 'Development Sandbox' containing '2' files at '9:12:22.275'.


Compilation ended at '9:12:38.754'.

System.IO.FileNotFoundException: Could not find file 'C:\sandbox\report.rdlc'.
File name: 'C:\sandbox\report.rdlc'
   at System.IO.FileStream.ValidateFileHandle(SafeFileHandle fileHandle)
   at System.IO.FileStream.CreateFileOpenHandle(FileMode mode, FileShare share, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize)
   at System.IO.File.ReadAllBytes(String path)
   at Microsoft.Dynamics.Nav.CodeAnalysis.RelativeFileSystem.ReadBytes(String path) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Compilation\RelativeFileSystem.cs:line 23
   at Microsoft.Dynamics.Nav.CodeAnalysis.Packaging.PackageModuleOutputter.AddLayoutDocuments() in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Packaging\PackageModuleOutputter.cs:line 35
   at Microsoft.Dynamics.Nav.CodeAnalysis.Packaging.PackageModuleOutputter.FinalizeModule() in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Packaging\PackageModuleOutputter.cs:line 31
   at Microsoft.Dynamics.Nav.CodeAnalysis.Compilation.WriteToOutput(ModuleOutputter outputter, ModuleBuilder moduleBuilder, ModuleSymbol compiledModuleSymbol, DiagnosticBag diagnostics) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\Compilation\Compilation.cs:line 893
   at Microsoft.Dynamics.Nav.CodeAnalysis.CommandLine.CommonCompiler.PackageTheModule(Compilation compilation, CompilerTextWriter consoleOutput, CancellationToken cancellationToken, ErrorLogger errorLogger, AnalyzerDriver analyzerDriver, Predicate`1 filterOpt, Predicate`1 radCompilationFilter) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\CommandLine\CommonCompiler.cs:line 728
   at Microsoft.Dynamics.Nav.CodeAnalysis.CommandLine.CommonCompiler.RunCoreImpl(CompilerTextWriter consoleOutput, ErrorLogger errorLogger, Compilation compilation, AnalyzerDriver analyzerDriver, IEnumerable`1 analyzerExceptionDiagnostics, IEnumerable`1 additionalTextFiles, CancellationToken cancellationToken) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\CommandLine\CommonCompiler.cs:line 529
   at Microsoft.Dynamics.Nav.CodeAnalysis.CommandLine.CommonCompiler.RunCore(CompilerTextWriter consoleOutput, ErrorLogger errorLogger, CancellationToken cancellationToken) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\CommandLine\CommonCompiler.cs:line 476
   at Microsoft.Dynamics.Nav.CodeAnalysis.CommandLine.CommonCompiler.Run(CompilerTextWriter consoleOutput, CancellationToken cancellationToken) in D:\a\1\s\source\Prod\Microsoft.Dynamics.Nav.CodeAnalysis\CommandLine\CommonCompiler.cs:line 332
   at Microsoft.Dynamics.Nav.CodeAnalysis.CommandLine.Alc.<>c__DisplayClass1_0.<Run>b__0(CompilerTextWriter textWriterOut, CompilerTextWriter _) in D:\a\1\s\source\Prod\alc\Alc.cs:line 37
   at Microsoft.Dynamics.Nav.CodeAnalysis.CommandLine.ConsoleUtil.RunWithOutput[T](Boolean useUtf8Encoding, Func`3 func) in D:\a\1\s\source\Prod\alc\ConsoleUtil.cs:line 25
   at Microsoft.Dynamics.Nav.CodeAnalysis.CommandLine.Alc.Run(String clientDirectory, String baseDirectory, String[] args, IAnalyzerAssemblyLoader analyzerLoader) in D:\a\1\s\source\Prod\alc\Alc.cs:line 37
E:\Repositories\developer-sandbox\output\appfile.app successfully created in 20 seconds
```

BcContainerHelper version: 1.0.12
Base image: `mcr.microsoft.com/dynamicsnav:10.0.17763.1577-generic`
Windwos version: 10.0.17763.1697 (Windows Server 2019 Version 1809)

**Steps to reproduce**
1. Create an empty AL Project
2. Create a report
3. Compile it with the command I gave above

**Comment by freddydk** [comment]
Could you open Compile-AppInNavContainer.ps1 in c:\program files\windowspowershell\modules\bccontainerhelper\version\apphandling and insert these three lines after the & .\alc.exe $alcParameters line:

```
        if (!$?)  {
            throw "App generation failed"
        }
```

This should throw an error if alc raises an errorlevel.

**Comment by michvllni** [comment]
Yes, after inserting that the error is thrown, thank you!
Is that something that is relevant for everyone or is it too specific?

**Comment by freddydk** [comment]
I think that if alc.exe throws an error - then compile should fail.
I will add this to the next preview and see if people experience problems.

**Comment by michvllni** [comment]
Perfect, thank you very much!

**Comment by freddydk** [comment]
Shipped in BcContainerHelper 2.0.6
